scripts/artifacts.py

```diff
@@ -0,0 +1,263 @@
+        ax.legend(bbox_to_anchor=(1.05, 1), loc="upper left")
+
+    plt.tight_layout()
+    plt.savefig(out)
+
+
+@cli.command()
+@click.option("--previous", required=True)
+@click.option("--next", required=True)
+def compare_stats(previous, next):
+    previous_stats = mk_commit_stats(fetch_commit_info(previous))
+    next_stats = mk_commit_stats(fetch_commit_info(next))
+
+    print_diffs(
+        mk_diffs(previous_stats.gas_costs, next_stats.gas_costs),
+        title="Gas costs",
+        header_previous=previous_stats.info.ref,
+        header_next=next_stats.info.ref,
+    )
+
+    print_diffs(
+        mk_diffs(previous_stats.entrypoint_sizes, next_stats.entrypoint_sizes),
+        title="Entrypoint sizes",
+        header_previous=previous_stats.info.ref,
+        header_next=next_stats.info.ref,
+    )
+
+
+# From the given $ref (usually a commit hash or a branch name), fetch the latest $limit commits
+# (from children to parents) and their workflow ids.
+def fetch_commit_infos(ref, *, limit):
+    query = """
+    query($ref:String!, $limit:Int!) {
+      repository(owner: "tezos-checker", name: "checker") {
+        object(expression: $ref) {
+          ... on Commit {
+            history(first: $limit) {
+              nodes {
+                message
+                oid
+                statusCheckRollup {
+                  contexts(first: 10) {
+                    edges {
+                      node {
+                        ... on CheckRun {
+                          checkSuite {
+                            workflowRun {
+                              databaseId
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  """
+
+    vars = {"ref": ref, "limit": limit}
+
+    ret = requests.post(
+        ENDPOINT_GRAPHQL, json={"query": query, "variables": vars}, headers=headers
+    )
+    ret.raise_for_status()
+    ret = ret.json()
+
+    commit_infos = [
+        CommitInfo(
+            message=get_first_line(commit["message"]),
+            ref=ref if i == 0 else f"{ref}~{i}",
+            rev=commit["oid"],
+            workflow_ids=[
+                check["node"]["checkSuite"]["workflowRun"]["databaseId"]
+                for check in commit["statusCheckRollup"]["contexts"]["edges"]
+            ]
+            if commit["statusCheckRollup"]
+            else [],
+        )
+        for i, commit in enumerate(
+            ret["data"]["repository"]["object"]["history"]["nodes"]
+        )
+        if commit["statusCheckRollup"]
+    ]
+
+    return commit_infos
+
+
+def fetch_commit_info(expr):
+    return fetch_commit_infos(expr, limit=1)[0]
+
+
+def mk_commit_stats(info):
+    stats = download_stats(info.workflow_ids)
+
+    gas_costs = stats.get("gas-costs", [{}])[0]
+    # TODO: Gas costs are strings for no reason.
```

Comment:
I think that this is just a quirk of the relevant RPC response on tezos. Perhaps we can also open an issue to make sure that we are converting these to ints in the end-to-end tests?

Reply:
Yeah, I will just send a PR to cast them to integers on end-to-end tests, and we can fix this one after that one is merged to master.